uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -1120,6 +1379,12 @@ private Optional<IdentityProvider<?>> readIdpViaDb(final String id, final String
         return Optional.of(idp);
     }
 
+    private IdentityProvider<?> updateIdpViaDb(final String zoneId, final IdentityProvider<?> idp) {
+        final JdbcIdentityProviderProvisioning identityProviderProvisioning = webApplicationContext
+                .getBean(JdbcIdentityProviderProvisioning.class);
+        return identityProviderProvisioning.update(idp, zoneId);
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.